### PR TITLE
scan_line() does not ignore print statements

### PR DESCRIPTION
**Pull request by sherrardb**
good day,
while using this wonderful tool to update and standardize `PREREQ_PM` across our many `Makefile.PL`, i discovered some instances of false positives, mostly involving printed strings with the word 'do' in them.

i have added some test cases as well as a proposed fix. but i consider my fix to be sub-par because it doesn't really get at the root of the issue, it simply avoids problems in some very specific cases. for example, it doesn't cover `say ...` or any other method one might use to render text. it also doesn't cover the failing text case that i included.

basically, i can see that the underlying issue is one of being able to parse certain syntax with a little more granularity, but i could not identify any clean way to make the relevant line(s) more robust based on my perusal of the existing code.

this PR was not intended to be a finished product, but i hope i have provided enough detail, context and clarity of intent. if not, please let me know.

### CODE REVIEW

**Comment by rschupp** [comment]
Thx, but this isn't the correct fix. I've reworked the recognition of `do STRING` though, you may try with current HEAD.
Note that `Module::ScanDeps` will never report dependencies 100% accurately. Its heuristics lean on the "include all possibilities" - that's what its main consumer `PAR::Packer` expects.

**Comment by sherrardb** [comment]
current `HEAD` works a treat WRT my reported issues. didn't know about feeding into `PAR::Packer`. thanks for the quick update and the context.
cheers
